### PR TITLE
Rename near to neard

### LINKED ISSUE #1141: near->neard
Rename near to neard, since it is a daemon and near is already used by the near-shell.
Also, there is a convention to name services and cli tools for them as <name>d and <name> respectively.

### PR DESCRIPTION
**Pull request by ailisp**
Rename near to neard.

Fixes #1141 

Test Plans
-------------
CI should pass, near binary should still be kept for compatibility.

### CODE REVIEW

File: neard/Cargo.toml [hunk 1]
```diff
@@ -48,3 +48,7 @@ testlib = { path = "../test-utils/testlib" }
 adversarial =  ["near-client/adversarial", "near-network/adversarial"]
 expensive_tests = ["near-client/expensive_tests", "near-epoch-manager/expensive_tests", "near-chain/expensive_tests"]
 metric_recorder = ["near-network/metric_recorder", "near-client/metric_recorder"]
+
+[[bin]]
+path = "src/near.rs"
+name = "near"
```
Comment (bowenwang1996) [inline neard/Cargo.toml]: shouldn't this be `neard` as well?

File: neard/src/near.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+use neard::neard_main;
```
Comment (bowenwang1996) [inline neard/src/near.rs]: what's the purpose of this file? It seems that we have two mains?

File: neard/Cargo.toml (hunk 1 above)
Comment (frol) [reply neard/Cargo.toml]: @bowenwang1996 I believe this is for the backward compatibility. We don't want to break all the tooling at once, thus we will have `near` and `neard` binaries in nearcore while we migrate our tooling.

Comment (frol) [reply neard/Cargo.toml]: @ailisp I wonder if we can use `path = "src/main.rs"` for this binary as well, this way we don't need to play around `lib.rs` + `main.rs` & `near.rs`.

Comment (ailisp) [reply neard/Cargo.toml]: @frol Doing that cargo will give a warning, and currently we have "fail on warning" on ci

Comment (ailisp) [reply neard/Cargo.toml]: Warning is 
`warning: /home/bo/workspace/nearcore2/neard/Cargo.toml: file found to be present in multiple build targets: /home/bo/workspace/nearcore2/neard/src/main.rs`
If doing:
```
[[bin]]
path = "src/main.rs"
name = "neard"

[[bin]]
path = "src/main.rs"
name = "near"
```
With only use 
```
[[bin]]
path = "src/main.rs"
name = "near"
```
cargo only compile it to near but no neard

Comment (frol) [reply neard/Cargo.toml]: We can workaround the warning my using:

```toml
[[bin]]
path = "src/main.rs"
name = "neard"

[[bin]]
path = "src/../src/main.rs"
name = "near"
```

This is a hack, but it is not as intrisuve as the need to have `lib.rs` + two `main.rs` and `near.rs`.

WDYT?

Comment (ailisp) [reply neard/Cargo.toml]: :joy: this is smart, I definitely won't report "cargo cannot detect `path = "src/../src/main.rs"` replicate with `path = "src/main.rs"`" to cargo issues

File: neard/src/lib.rs [hunk 3]
```diff
@@ -49,7 +153,7 @@ fn init_logging(verbose: Option<&str>) {
         .init();
 }
 
-fn main() {
+pub fn neard_main() {
```
Comment (frol) [inline neard/src/lib.rs]: I think we can bring back it to the `main.rs`. I expect the changes to `lib.rs` and `main.rs` to be mostly reverted.

Comment (ailisp) [reply neard/src/lib.rs]: OK

**Pull request merged**